### PR TITLE
Install.sh improvements

### LINKED ISSUE #13: Improve the check for "should we stamp pulumi into the user's bashrc"
Currently we decide whether or not to add `export PATH=\$PATH:\$HOME/.pulumi/bin` to the user's `.bash{rc,_profile}` by searching for that precise line. This will not work if the user has manually added Pulumi to the path/edited that line/etc. It might be better to just check to see if pulumi is on the path to begin with or to instead search for our comment (`# add Pulumi to the PATH`).

### PR DESCRIPTION
**Pull request by ellismg**
1. If `pulumi` is already on the path, don't try add anything to the
   user's profile.

2. If we are trying to update the user's profile, look for our message
   that says it adds Pulumi to the path instead of looking for our
   exact command to modify `$PATH`

3. Print an a link to slack in the case where the script fails.

Fixes #13
Fixes pulumi/pulumi#1489

### CODE REVIEW

File: dist/install.sh [hunk 1]
```diff
@@ -79,48 +91,54 @@ if curl --fail -L -o "${TARBALL_DEST}" "${TARBALL_URL}"; then
...
-        # file when trying to put our export into a profile. On *NIX, .bashrc is
-        # prefered as it is sourced for new interactive shells.
-        if [ "$(uname)" != "Darwin" ]; then
-            if [ -e "${HOME}/.bashrc" ]; then
-                PROFILE_FILE="${HOME}/.bashrc"
-            elif [ -e "${HOME}/.bash_profile" ]; then
-                PROFILE_FILE="${HOME}/.bash_profile"
+# Now that we have installed Pulumi, if it is not already on the path, let's add a line to the
+# user's profile to add the folder to the PATH for future sessions.
+if ! command -v pulumi >/dev/null; then
```
Comment (pgavlin) [inline dist/install.sh]: Nit: should we ensure that this is the `pulumi` we just installed?

Comment (ellismg) [reply dist/install.sh]: I thought about this, but I'm not sure it is worthwhile.  Let's assume you already had a copy of `pulumi` on your path somewhere and it was in a different location than `$HOME/.pulumi/bin`.  So what do we do in this case?

If we add something to your .bashrc, then is probable that future invocations of your shell are going to prefer this new `pulumi` to the existing one you have, given that we add our line at the bottom of your `.bashrc`.  This seems more surprising to me.  Put another way, it's not clear to me that we can figure out the correct user intent and so I'm happy to just not do any additional work.  

If we want to add a warning along the lines of "It looks like you already had pulumi on your path at `<LOCATION>`, you may need to add `$HOME/.pulumi/bin` to your path explicitly to use the just installed version", I'd be fine with that.  What do you think?

Comment (pgavlin) [reply dist/install.sh]: > I thought about this, but I'm not sure it is worthwhile. Let's assume you already had a copy of pulumi on your path somewhere and it was in a different location than $HOME/.pulumi/bin. So what do we do in this case?
> ...
> If we want to add a warning along the lines of "It looks like you already had pulumi on your path at <LOCATION>, you may need to add $HOME/.pulumi/bin to your path explicitly to use the just installed version", I'd be fine with that. What do you think?

I like this idea.

Comment (ellismg) [reply dist/install.sh]: > I like this idea.

Done, PTAL

**Pull request merged**